src/tests/p11test/p11test_case_common.c

```diff
@@ -1213,11 +1213,23 @@ const char *get_mechanism_flag_name(int mech_id)
 		case CKF_EC_ECPARAMETERS:
 			return "CKF_EC_ECPARAMETERS";
 		default:
-			sprintf(flag_buffer, "0x%.8X", mech_id);
-			return flag_buffer;
+			return "unknown";
 	}
 }
 
+const char *
+get_mechanism_all_flag_name(int mech_id)
+{
+	CK_FLAGS j;
+
+	flag_buffer[0] = '\0';
+	for (j = 1; j <= CKF_EC_COMPRESS; j = j << 1)
+		if ((mech_id & j) != 0)
+			snprintf(flag_buffer + strlen(flag_buffer), sizeof(flag_buffer) - strlen(flag_buffer), "%s,", get_mechanism_flag_name(j));
+	flag_buffer[strlen(flag_buffer) - 1] = '\0';
```

Comment:
I think this line is not needed. The `snprintf` always terminate the string with null byte so here you only override that NULL byte (as `strlen` already searches for the NULL byte).